src/Debugger/Test/DebugSessionExtensions.cs

```diff
@@ -23,7 +23,7 @@ internal static class DebugSessionExtensions {
             EventHandler<DebugBrowseEventArgs> handler = null;
             handler = (sender, e) => {
                 session.Browse -= handler;
-                browseTcs.SetResult(true);
+                browseTcs.TrySetResult(true);
```

Comment:
EventTaskSource waits for only one event notification. Do you mean that here you need to wait for 1 or 2?